node-fs-readfile/cat.js

```diff
@@ -0,0 +1,22 @@
+import { readFile } from 'node:fs/promises';
+
+async function readCatAnyFiles() {
+  const fileArray = process.argv.slice(2);
+
+  const promiseArray = fileArray.map(async (file) => {
+    const filePath = new URL(file, import.meta.url);
+    const contents = await readFile(filePath, { encoding: 'utf8' });
+    return contents;
```

Comment:
async/await here works, but it's not necessary. `readFile` returns a Promise, which is what you want to pass to Promise.all.

Also, although the Node documentation may look like you need import.meta.url, etc., it actually isn't necessary. By default, `readFile` will look in the current directory for the file.

So you can just use `fileArray.map((file) => readFile(file, 'utf8'))`.

(Also note that as a shortcut, `readFile` allows you to send the encoding as a string in the second parameter.)

Reply:
Updated